src/main/java/org/reactome/addlinks/fileprocessors/HmdbMetabolitesFileProcessor.java

```diff
@@ -130,4 +129,54 @@ public HmdbMetabolitesFileProcessor()
 		return null;
 	}
 
+	
+	private void transformXmlToTsv(String inputFilename, String outfileName)
+	{
+		
+		try(OutputStream outStream = new FileOutputStream(outfileName);)
+		{
+			XMLInputFactory xif = XMLInputFactory.newInstance();
+			XMLStreamReader xsr = xif.createXMLStreamReader(new FileReader(inputFilename));
+			xsr.nextTag();
+			Source source = new StreamSource(this.getClass().getClassLoader().getResourceAsStream("hmdb_metabolites_transform.xsl"));
+			TransformerFactory factory = TransformerFactory.newInstance();
+			Transformer transformer = factory.newTransformer(source);
+			
+			while (xsr.nextTag() == XMLStreamConstants.START_ELEMENT)
+			{
+				if (xsr.getName().getLocalPart().equals(METABOLITE_ELEMENT_NAME))
+				{
+					StAXSource src = new StAXSource(xsr);
+					Result result = new StreamResult(outStream);
+					transformer.transform(src, result);
+				}
+			}
+			xsr.close();
+		}
+		catch (FileNotFoundException e ) 
+		{
+			logger.error("Input XML file was not found!", e);
+			e.printStackTrace();
+		}
+		catch (XMLStreamException e)
+		{
+			logger.error("Error streaming XML file!", e);
+			e.printStackTrace();	
+		}
+		catch (TransformerConfigurationException e)
+		{
+			logger.error("Error with XSL file!", e);
+			e.printStackTrace();
+		}
+		catch (TransformerException e)
+		{
+			logger.error("Data transformation error!", e);
+			e.printStackTrace();
+		}
+		catch (IOException e)
+		{
+			logger.error("I/O Error", e);
+			e.printStackTrace();
+		}
```

Comment:
This is a lot of code in the method body for exception handling.  Since the only thing that differs is the logged error message, could you do this:

```
try {
    ...
} catch (FileNotFoundException | XMLStreamException | TransformerException | IOException e) {
    // Or use a helper method call to get the String you want
    logger.error("Error due to " + e.getClass().getName() + " exception", e);
    e.printStackTrace();
}
```
You can just have a generic message since the exception type will be included in the log or you can have a method like 'private String getExceptionMessage(Exception e)` that returns the String error message you want to log without distracting from the main logic.

Reply:
I like having more specific messages. Gives the reader of the message a bit more of a clue than a generic message.

Reply:
That's fine - you can use a helper method to return the specific message you want.  I just think over 20 lines to handle exceptions is a bit much.

Reply:
I've removed the `e.printStackTrace()` calls (I don't think they were necessary, since the overload of the `error()` method should _log_ the full stack trace), but I'm still not in favour of a helper method here.

Reply:
Is there a reason you're opposed to a helper method?  

I think it would be useful - it allows you to collapse the exception handling into one catch block and reduce the size of the method.  I think it makes sense to try to keep the method concise and as focused on its main purpose as possible.  Using a helper method would help reduce the footprint of the exception handling which is incidental to the method's purpose.  For me, it really helps readability and understanding.

Reply:
In this particular case, `FileNotFoundException` is a subtype of `IOException` so the expression `catch (FileNotFoundException | XMLStreamException | TransformerException | IOException e)` causes problems because `FileNotFoundException` _is_ an `IOException`. If you want to give them separate handling (and I did because they are not thrown by the same message so I wanted different messges), they need separate `catch`es. So I'd have to have at least two separate catches - one for `FileNotFoundException`, the other for the remaining three that are handled by a helper method, which I personally would find confusing and weird. Or, lose the distinction between `FileNotFound...` and `IO...`.

AND... there's a similar problem with `TransformerConfigurationException` and `TransformerException`... I prefer to keep them separate so I can give a specific messages because one exception is caused by the transformation itself, and the other is caused by issues within the XSL file.

So... in order to keep the 5 distinct messages, I still need 3 `catch`-blocks:
```Java
catch (FileNotFoundException e)
{
    logger.error("Could not read input file", e);
}
catch (XMLStreamException| TransformerConfigurationException | IOException e) 
{
    logXml2TsvException(e);
}
catch (TransformerException e)
{
    logger.error("Data transformation error", e);
}
```
And the handler will only handle `XMLStreamException`, `TransformerConfigurationException`, `IOException`.

---
In general, I think that 4 or 5 or 6 `catch`-blocks with _simple_ exception-handling code is really not at all unreasonable. In this case, I think adding another method to handle all exceptions will increase overall complexity - now the method needs to have its own logic to determine which exception type it received so it knows what to do whereas the original try/catch had that logic built in.

Reply:
Okay, I see what you're saying.  We can leave it.